src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1

```diff
@@ -2370,4 +2370,116 @@ function Test-VirtualMachineScaleSetAutoRepair
+		$inboundNatPool = New-AzLoadBalancerInboundNatPoolConfig -Name $inboundNatPoolName -FrontendIPConfigurationId `
+			$frontend.Id -Protocol Tcp -FrontendPortRangeStart 3360 -FrontendPortRangeEnd 3368 -BackendPort 3370;
+		$lbrule = New-AzLoadBalancerRuleConfig -Name $lbruleName `
+			-FrontendIPConfiguration $frontend -BackendAddressPool $backendAddressPool `
+			-Probe $probe -Protocol Tcp -FrontendPort 80 -BackendPort 80 `
+			-IdleTimeoutInMinutes 15 -EnableFloatingIP -LoadDistribution SourceIP;
+		$actualLb = New-AzLoadBalancer -Name $lbName -ResourceGroupName $rgname -Location $loc `
+			-FrontendIpConfiguration $frontend -BackendAddressPool $backendAddressPool `
+			-Probe $probe -LoadBalancingRule $lbrule -InboundNatPool $inboundNatPool;
+		$expectedLb = Get-AzLoadBalancer -Name $lbName -ResourceGroupName $rgname
+ 
+		# New VMSS Parameters
+		$vmssName = 'vmss' + $rgname;
+		$adminUsername = 'Foo12';
+		$adminPassword = $PLACEHOLDER;
+ 
+		#$PublisherName = "MicrosoftWindowsServer" 
+		#$Offer         = "WindowsServer" 
+		#$Sku           = "2012-R2-Datacenter" 
+		#$Version       = "4.127.20190204"
+		$imgRef = Get-DefaultCRPImage -loc $loc;
+		$vhdContainer = "https://" + $stoname + ".blob.core.windows.net/" + $vmssName;
+ 
+		$ExtName = "CSETest";
+		$Publisher = "Microsoft.Compute";
+		$ExtType = "BGInfo";
+		$ExtVer = "2.1";
+ 
+		$ipCfg = New-AzVmssIPConfig -Name 'test' `
+			-LoadBalancerInboundNatPoolsId $expectedLb.InboundNatPools[0].Id `
+			-LoadBalancerBackendAddressPoolsId $expectedLb.BackendAddressPools[0].Id `
+			-SubnetId $subnetId;
+ 
+		$vmss = New-AzVmssConfig -Location $loc -SkuCapacity 2 -SkuName 'Standard_E4-2ds_v4' -UpgradePolicyMode 'Automatic' `
+			| Add-AzVmssNetworkInterfaceConfiguration -Name 'test' -Primary $true -IPConfiguration $ipCfg `
+			| Add-AzVmssNetworkInterfaceConfiguration -Name 'test2' -IPConfiguration $IPCfg `
+			| Set-AzVmssOSProfile -ComputerNamePrefix 'test' -AdminUsername $adminUsername -AdminPassword $adminPassword `
+			| Set-AzVmssStorageProfile -Name 'test' -OsDiskCreateOption 'FromImage' -OsDiskCaching 'None' `
+			-ImageReferenceOffer $imgRef.Offer -ImageReferenceSku $imgRef.Skus -ImageReferenceVersion $imgRef.version `
+			-ImageReferencePublisher $imgRef.PublisherName -VhdContainer $vhdContainer `
+			| Add-AzVmssExtension -Name $ExtName -Publisher $Publisher -Type $ExtType -TypeHandlerVersion $ExtVer -AutoUpgradeMinorVersion $True
+ 
+		Start-Transcript -Path "transcript.txt"
+		#create the vmss 
+		New-AzVmss -ResourceGroup $rgname -Name $vmssName -VirtualMachineScaleSet $vmss
+		Stop-Transcript
+ 
+		$wordToFind="You are deploying VMSS pinned to a specific image version from Azure Marketplace.";
+		#$wordToFind ="dummy string";
+		$file = Get-Content -path "transcript.txt";
+		$containsWord = $file | %{$_ -match $wordToFind}
+		If($containsWord -contains $true)
```

Comment:
If there is a line change in the terminal, this test may not work even though the warning message is returned.

Reply:
updated to join newlines from the transcript files.